File: protocol-generator/typescript/test/host.ts
/**
 * entity-core-protocol-typescript — the bare peer host.
 *
 * `runHost(argv, () => {})` and nothing else: the runnable target for S4 conformance
 * (`run-s4.sh`) and the keystone peer contract's BARE HOST. Every flag, the readiness
 * record and the stop behaviour are {@link runHost}'s — see `src/host.ts`:
 *
 *   --port N  --bind ADDR  --name NAME  --validate  --seed-policy PATH
 *   --max-frame-bytes N  --ready-file PATH  --debug-open-grants (deprecated)  --help
 *
 * Without `--name` a fixed test seed (0x11 × 32) is used, so the peer id is stable.
 * One `LISTENING {"record":"keystone-peer-ready/1",…}` line goes to stdout once the
 * peer is listening; a harness waits for it.
 *
 * Run (in-container, after tsc): `node dist/test/host.js --port 7777 [--name NAME] [--seed-policy PATH] [--validate]`.
 */

import { runHost } from "../src/index.js";

runHost(process.argv.slice(2), () => {}).then(
  (code) => process.exit(code),
  (err: unknown) => {
    process.stderr.write(`fatal: ${err instanceof Error ? (err.stack ?? err.message) : String(err)}\n`);
    process.exit(1);
  },
);

File: protocol-generator/python/src/entity_core/host.py
"""Standalone entity-core-protocol-python peer host (``python -m entity_core.host``,
console script ``entity-core-peer``) — the S4 conformance host ``validate-peer`` dials, and
the keystone peer contract's BARE HOST.

It is ``run_host(argv, no-op)``: every flag, the readiness record and the stop behaviour
live in :mod:`entity_core.peer.host`, so a composing program that calls
``run_host(argv, configure)`` is this host plus its ``configure`` and nothing else
(keystone peer contract ``embed.host_main``).  See that module for the flag set.

On startup it prints ``LISTENING <readiness-record JSON>`` on stdout, then serves until
SIGTERM / SIGINT.
"""

from __future__ import annotations

import sys

from .peer.host import run_host


def main(argv: list[str] | None = None) -> int:
    return run_host(sys.argv[1:] if argv is None else argv, None)


if __name__ == "__main__":
    raise SystemExit(main())
